### ISSUE TITLE
IN public cloud byos systems and hybrid system updates should be synced to SCC

### ISSUE THREAD

**Issue by rjschwei** [SUSE/rmt#1205]
When a BYOS system or a system using an add on product uses he Public Cloud update infrastructure the information of these systems needs to be passed on to SCC. BYOS systems and hybrid systems consume a SUSE subscription in the same way it is consumed in the data center, i.e. we can consider these systems as "direct registrations". As such during the automatic syncing of RMT data to SCC these systems need to be synced even if data forwarding is disabled.

Note that a translation has to happen between the credentials used for RMT repo syncing and the customer credentials that were used when the BYOS system or Hybrid system was registered against RMT.

**Comment by jesusbv** [comment]
OK, I understand we do not treat `hybrid` as a `payg` until an add-on with a regcode is added but keeps its `payg` behavior 

If that is the case (meaning `hybrid` behaves exactly as `byos`), I will do implement the changes right away

Question, a `hybrid` system, AFAIU, it is not always a `hybrid` system, 
meaning....a user could have a `payg` system running and 
at some point decide to activate an extension with a regcode making the system `hybrid`, 
I understand that's still valid ?

Otherwise, please correct me or clarify what's wrong/incomplete

**Comment by jesusbv** [comment]
@rjschwei ^^ can you confirm ?

**Comment by rjschwei** [comment]
Correct, a system only becomes a hybrid system when a user activates a non-free extension. Once the user activates the non-free extension that part of the system is basically BYOS and therefore needs to be reported/synced to SCC.

A BYOS system is no different than a system that runs in a data canter registered to an RMT server running in the data center. If I remember correctly we do not allow/support customers to set the flag for their own private RMT servers. @digitaltom do I remember this correctly?

**Comment by jesusbv** [comment]
OK, so currently, 

- a PAYG system gets announce to SCC when a non free extensions is activated (the extension gets announce to SCC as well)
- the now HYBRID system gets de register from SCC when user runs `cleanup` 
- the non free extension gets removed from SCC when user runs `SUSEConnect -d -p <non-free-extension>`
- when the non free extension that made the PAYG system HYBRID gets removed from SCC, we do not actually change the system back to PAYG (maybe we should, current decision is not to)
- if the system is PAYG or HYBRID, and user tries to activate a product without a regcode, [it will check system's current subscription](https://github.com/SUSE/rmt/pull/1203/files#diff-7edbe8015a11c2bffc38e12ec3aa8e97c68e66201ef3bf64947d3bad213ea1e1R61)

am I missing something that should be implemented and it is not to fix this issue ? 

@rjschwei ^

**Comment by digitaltom** [comment]
With "automatic syncing of RMT data to SCC", I think @rjschwei refers to https://github.com/SUSE/rmt/blob/master/lib/rmt/scc.rb#L77.
This method is called by `rmt-cli sync systems` and syncs the latest system changes to the configured customer account in SCC. For SUSE public cloud RMTs, this is skipped by the `Settings.scc.sync_systems = false` setting, because we don't want all pubcloud customer systems (neither payg, byos, hybrid) to show up in the "SUSE Public Cloud RMT Infra (Production) A201502" organization in SCC. 

For MSPs that are using the pubcloud engines, I think payg+hybrid systems could get synced to SCC, because we bill the MSP provider for those?

**Comment by jesusbv** [comment]
>  I think @rjschwei refers to

OK, so you are not sure either ? I will address what you explained in the comment
hopefully that's it otherwise, guess it will get clarified
